apps/shell/app.js

```diff
@@ -55,9 +56,67 @@ app.use(process.env.PASSENGER_BASE_URI || '/', router);
 var server = new http.createServer(app);
 var wss = new WebSocket.Server({ noServer: true });
 
+
+
+whitelist = process.env.SSHHOST_WHITELIST;
+if (whitelist === undefined){
+  whitelist = [];
+} else {
+  whitelist = whitelist.split(':');
+}
+
+var filter = '.yml'; // add regex capability
+var files = fs.readdirSync('/etc/ood/config/clusters.d');
+files.forEach( function (file){
+  if (file.indexOf(filter) >= 0){
+    var fullPath = '/etc/ood/config/clusters.d/' + file;
+    var data = yaml.safeLoad(fs.readFileSync(fullPath)); //data is in json format
+    var errorMessage = 'Cluster config file does not follow correct v2 schema. Please see osc.github.io/ood-documentation.';
+
+    if (data.hasOwnProperty('v2')){
```

Comment:
@matthu017 Take a look at optional chaining in JS https://v8.dev/features/optional-chaining. Using optional chaining here would eliminate the nested if statements

Reply:
That is cool! But optional chaining doesn't work in Edge 79 released in Jan: https://caniuse.com/#feat=mdn-javascript_operators_optional_chaining

Reply:
@msquee corrected me offline. This is all server side code we are dealing with :-). If optional chaining works with our node it is the right solution.